res/controllers/Vestax-VCI-100MKII-scripts.js

```diff
@@ -257,32 +257,53 @@ VCI102.next_effect = function(ch, midino, value, status, group) {
     }
 };
 
-VCI102.loopSize = [4, 4, 4, 4];  // saved beatloop_size
-
 VCI102.loop = function(ch, midino, value, status, group) {
     if (value) {
         if (engine.getValue(group, "loop_enabled")) {
             engine.setValue(group, "reloop_toggle", 1);
-            // restore beatloop_size
-            engine.setValue(group, "beatloop_size", VCI102.loopSize[ch]);
         } else {
-            VCI102.loopSize[ch] = engine.getValue(group, "beatloop_size");
             engine.setValue(group, "beatloop_activate", 1);
+            engine.setValue(group, "beatloop_activate", 0);
         }
     }
 };
 
 VCI102.reloop = function(ch, midino, value, status, group) {
     if (value) {
         if (engine.getValue(group, "loop_enabled")) {
+            engine.setValue(group, "loop_out", 1);
```

Comment:
This controller sends different messages depending on the shift button. I map VCI102.loop (without shift) and VCI102.reloop (with shift) to the same button 27. So total behavior is,
* Push with no loop active: beatloop_activate
* Push with loop active: reloop_toggle
* Shift + push with no loop active: reloop_toggle
* Shift + push with loop active: loop_out

Of course I have realized that "loop_out" is intended to be used with "loop_in" for manual loop, but it can be used with "beatloop_active" (or other beatloop starters) for manual loop. For example,
* Begin beatloop by "beatloop_active" with size 8
* After 5 beats, activate-deactivate "loop_out"
* Continue to loop with size 5

Though it is not intended use of "loop_out", but works fine at least from Mixxx 1.11 till now. This use of "loop_out" can add manual loop to usual beatloop mapping with minimal cost, I think.